hypothesis_crosshair_provider/crosshair_provider.py

```diff
@@ -60,22 +64,20 @@ def post_test_case_context_manager(self):
             StateSpaceContext(self._previous_space),
             COMPOSITE_TRACER,
         ):
-            # IgnoreAttempt is possible here in theory, but hopefully won't happen because we've
-            # already fixed the drawn values
             yield
 
     def _make_statespace(self):
         options = DEFAULT_OPTIONS.overlay(AnalysisOptionSet(analysis_kind=[]))
+        # TODO: is this the right way to get current settings?:
+        hypothesis_deadline = settings().deadline
```

Comment:
Yep, that's the best way to do it.  (and my apologies for the whole settings system, it's... messier than I'd like)

Reply:
Shrug! Seems fine; I don't know how to do it right in CrossHair either.